tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php

```diff
@@ -471,6 +587,35 @@ public function editBrandAddressWithFollowingDetails($manufacturerAddressReferen
         $editManufacturerAddressCommand->setCity($manufacturerAddress->getCity());
         $editManufacturerAddressCommand->setManufacturerId($manufacturerAddress->getManufacturerId());
         $editManufacturerAddressCommand->setCountryId($manufacturerAddress->getCountryId());
+        $editManufacturerAddressCommand->setStateId($manufacturerAddress->getStateId());
         $this->getCommandBus()->handle($editManufacturerAddressCommand);
     }
+
+    /**
+     * @param int $countryId
+     * @param array $testCaseData
+     *
+     * @return int|null
+     *
+     * @throws RuntimeException
+     */
+    private function getStateId(int $countryId, array $testCaseData): ?int
+    {
+        if (empty($testCaseData['State'])) {
+            return null;
+        }
+        $stateId = null;
+        /* @var CountryStateByIdChoiceProvider $countryStateChoiceProvider */
+        $countryStateChoiceProvider = $this->getContainer()->get('prestashop.adapter.form.choice_provider.country_state_by_id');
```

Comment:
Funny way to retrieve state ID 😄  I think it would be accepted here to use legacy class
Just suggestion, not blocking

Reply:
I admit it's not orthodox ^^ I only used the existing code and refacto it a little actually But it has the advantage to make sure you use a state that belongs to the country (which means our behat tests are actually smarter than our handlers.. 😅)